packages/core/src/tools/sanitize.ts

```diff
@@ -0,0 +1,235 @@
+  // We need to use source as sourceToSanitize could be a reference to a new object
+  // At this stage, we know the source is an object type
+  const sourceAsObject = source as object
+  if (visitedObjectsWithPath.has(sourceAsObject)) {
+    return `[Reference seen at ${visitedObjectsWithPath.get(sourceAsObject)!}]`
+  }
+
+  // Add processed source to queue
+  const currentPath = key ? `${parentPath}.${key}` : `${parentPath}`
+  const target = Array.isArray(sourceToSanitize) ? [] : {}
+  visitedObjectsWithPath.set(sourceAsObject, currentPath)
+  queue.push({ source: sourceToSanitize, target, path: currentPath })
+
+  return target
+}
+
+/**
+ * Used to approximate the number of characters elements occupy inside a container
+ * "key":"value" => returns a size of 13 if first value in collection (separatorLength=0), 14 otherwise (separatorLength=1)
+ *
+ * LIMITATIONS
+ * - Actual byte count may differ according to character encoding
+ *
+ * @param value              value in key/value pair
+ * @param separatorLength    0 if first pair in container, 1 otherwise (accounts for the comma separator)
+ * @param key                key in key/value pair (undefined for arrays)
+ *
+ */
+function computeSize(value: ContextValue, separatorLength = 0, key?: string) {
+  try {
+    return JSON.stringify(value).length + separatorLength + (key ? key.length + 3 : 0)
+  } catch {
+    // We do not expect this to happen as complex types will have been processed previously
+  }
+  return 0
+}
+
+/**
+ * Handles sanitization of simple, non-object types
+ *
+ */
+function sanitizePrimitivesAndFunctions(value: ExtendedContextValue) {
+  // BigInt cannot be serialized by JSON.stringify(), convert it to a string representation
+  if (typeof value === 'bigint') {
+    return `[BigInt] ${value.toString()}`
+  }
+  // Functions cannot be serialized by JSON.stringify(). Moreover, if a faulty toJSON is present, it needs to be converted
+  // so it won't prevent stringify from serializing later
+  if (typeof value === 'function') {
+    return `[Function] ${value.name || 'unknown'}`
+  }
+  // JSON.stringify() does not serialize symbols. We cannot use (yet) symbol.description as it is part of ES2019+
+  if (typeof value === 'symbol') {
+    return `[Symbol] ${value.toString()}`
+  }
+
+  return value
+}
+
+/**
+ * Handles sanitization of object types
+ *
+ * LIMITATIONS
+ * - If a class defines a toStringTag Symbol, it will fall in the catch-all method and prevent enumeration of properties.
+ * To avoid this, a toJSON method can be defined.
+ * - IE11 does not return a distinct type for objects such as Map, WeakMap, ... These objects will pass through and their
+ * properties enumerated if any.
+ *
+ */
+function sanitizeObjects(value: object) {
+  try {
+    // Handle events - Keep a simple implementation to avoid breaking changes
+    if (value instanceof Event) {
+      return {
+        type: value.type,
+        isTrusted: value.isTrusted,
+      }
+    }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏could we remove this for now and discuss strategies to serialize specific types later?

Reply:
To avoid breaking changes, we need to keep `{ isTrusted: true|false }`